File: Codeforces/1520/B.kt
fun main() {
    val arr = MutableList(0) { 0 }
    for (cnt in 1..9) {
        var ch = '1'
        while (ch <= '9') {
            val s = ch.toString().repeat(cnt)
            arr.add(s.toInt())
            ch++
        }
    }
    arr.sort()
    val T = readLine()?.toInt() ?: 1
    for (t in 1..T) {
        val n = readLine()?.toInt() ?: 1
        var (lo, hi, ans) = arrayOf(0, arr.size - 1, 0)
        while (lo <= hi) {
            val mid = lo + (hi - lo) / 2
            if (arr[mid] <= n) {
                ans = mid
                lo = mid + 1
            } else {
                hi = mid - 1
            }
        }
        println(ans + 1)
    }
}

File: Codeforces/1520/B.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t, n;
    cin >> t;
    vector<int> arr;
    for (int cnt = 1; cnt <= 9; ++cnt) {
        for (char ch = '1'; ch <= '9'; ++ch) {
            string s(cnt, ch);
            arr.push_back(stoi(s));
        }
    }
    sort(arr.begin(), arr.end());
    while (t--) {
        cin >> n;
        int lo = 0, hi = arr.size() - 1, ans = 0;
        while (lo <= hi) {
            int mid = lo + (hi - lo) / 2;
            if (arr[mid] <= n) {
                ans = mid;
                lo = mid + 1;
            } else {
                hi = mid - 1;
            }
        }
        cout << ans + 1 << "\n";
    }
    return 0;
}